### DISCUSSION TITLE
Does it handle redirects?

### DISCUSSION THREAD

**Discussion by Yahav** [PaperStrike/Pjax#289] [Q&A]
So i have a form that if the validation fails will return a redirect to the previous page with the validation errors in a session variable.
I noticed that on redirect it actually redirect the entire page, is there anything to do about that?

**Comment by Yahav** [comment]
Worth to mention that its a redirect from a POST to get.
Attached is a screenshot of what's happening on said redirect
![R3IwPGkS7x](https://user-images.githubusercontent.com/113236/163679277-8cf1b455-e7fe-4ff3-a1b8-13cc2f42859c.png)
And a screenshot of the events triggered
![9twJ2Zf0AC](https://user-images.githubusercontent.com/113236/163679318-12dd8cd1-b70b-4258-9b9b-33047cbd9a5b.png)

**Comment by PaperStrike** [comment]
HTTP Redirections are all handled in `fetch` by the browser, Pjax only parses the final document without even knowing how many redirections was made - just like you see in the event logs, the `response` is always the final 200. The `redirected` property will be `true` but it won't make any difference as it is of no use to Pjax currently.

And ah, Pjax seems to work fine in your screenshots. If it chose to make a full reload, it won't fire a `pjax:success` event, and the console is usually cleaned by the browser if a reload was successfully made. If the page is only visually reloaded, I suggest checking if your `selectors` option being too broad like selecting the whole `body`, or if any other script in the refreshing part is the one affecting the entire page.

**Reply by PaperStrike** [reply]
An example for how `fetch` works and how Pjax uses it:

```mermaid
sequenceDiagram
    Pjax->>Browser: Call fetch with a request to path 1
    Browser->>Server: Send the request to path 1
    loop HTTP Redirections
        Server-->>Browser: Send a redirect response (e.g., 302) to redirect to path N
        Browser-->>Server: Resend the request to path N
    end
    Server->>Browser: Send a 200 response
    Browser->>Pjax: Resolve the fetch with the 200 response
    Pjax->>Pjax: Parse the document in the response
```

It shouldn't matter if the request is a GET or a POST, redirected or unredirected, as long as the final response comes with a valid document.

**Comment by Yahav** [comment]
Thanks.
Yeah, since Fetch is handling the requests/redirects that shouldn't be a pjax issue.
Turned out it was a form validation js lib i have which submitted the actual form.
